File: Linear-Search/c/linearsearch.c
#include <stdio.h>
#include <assert.h>

int linearSearch(int numList[], int arraySize, int number);

int main(void) {
    int numbers[] = {1,2,3,4,5,6,7,8};
    int findNum = 4;

    assert(linearSearch(numbers, sizeof(numbers), findNum) == 0);
    printf("Number %d was found\n", findNum);
}

int linearSearch(int numList[], int arraySize, int number) {
    for (int i = 0; i < arraySize; i++) {
        if (numList[i] == number) {
            return 0;
        }
    }
    return 1;
}

File: Linear-Search/python/linearsearch.py
def linearSearch(number, numberList):
    for i in range(len(numberList)): 
        if numberList[i] == number:
            return i
    return -1

numberList = [1,2,3,4,5,6,7,8]
number = 4

assert linearSearch(number, numberList) == 3, 'Result shouild be 3'
print(f"{number} was found");

number = 12
assert linearSearch(number, numberList) == -1, 'Result shouild be -1'
print(f"{number} not found");

